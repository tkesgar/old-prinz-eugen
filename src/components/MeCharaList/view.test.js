import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import MeCharaListView from './view'

describe('MeCharaListView', () => {
  const wrapper = mount(MeCharaListView, {localVue})

  test('should mount successfully', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('MeCharaListView with empty list (default)', () => {
  const wrapper = mount(MeCharaListView, {localVue})

  test('should show empty message', () => {
    expect(wrapper.find('.chara-list-empty').exists()).toBe(true)
  })
})
