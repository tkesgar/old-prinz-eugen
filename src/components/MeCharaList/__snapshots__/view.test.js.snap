// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeCharaListView should match snapshot 1`] = `
<div
  class="chara-list-empty text-center"
>
  <p>
    Anda tidak memiliki karakter.
  </p>
   
  <p>
    Silakan buat karakter baru 
    <a
      class=""
      href="/chara/new"
      target="_self"
    >
      di sini
    </a>
    .
  </p>
</div>
`;
